tests/tests_pytorch/trainer/flags/test_val_check_interval.py

```diff
@@ -127,9 +130,110 @@ def test_val_check_interval_float_with_none_check_val_every_n_epoch():
     """Test that an exception is raised when `val_check_interval` is set to float with
     `check_val_every_n_epoch=None`"""
     with pytest.raises(
-        MisconfigurationException, match="`val_check_interval` should be an integer when `check_val_every_n_epoch=None`"
+        MisconfigurationException,
+        match=re.escape(
+            "`val_check_interval` should be an integer or a time-based duration (str 'DD:HH:MM:SS', "
+            "datetime.timedelta, or dict kwargs for timedelta) when `check_val_every_n_epoch=None`."
+        ),
     ):
         Trainer(
             val_check_interval=0.5,
             check_val_every_n_epoch=None,
         )
+
```

Comment:
it seems neither of these tests are checking that the time based validation works for all 3 input formats (string, dict, timedelta). Consider adding a test for `_parse_time_interval_seconds` or to parameterize one of the tests for the different formats